src/main/java/frc/robot/TestMode.java

```diff
@@ -0,0 +1,193 @@
+  public boolean climbTest;
+  public boolean algaeRollersForwardsTest;
+  public boolean algaeRollersBackwardsTest;
+  public boolean algaeIntakeUpTest;
+  public boolean algaeIntakeDownTest;
+  public boolean armElevatorTest;
+  public boolean armTest;
+  private final IndexerSubsystem indexerSubsystem = new IndexerSubsystem();
+  private final GamePieceManipulatorSubsystem gamePieceManipulatorSubsystem = new GamePieceManipulatorSubsystem();
+  private final ClimbSubsystem climbSubsystem = new ClimbSubsystem();
+  private final AlgaeSubsystem algaeSubsystem = new AlgaeSubsystem();
+  private final ArmSubsystem armSubsystem = new ArmSubsystem();
+
+  public TestMode() {
+
+  }
+
+  public Command testCommand() {
+    return testIndexerForwardsCommand().andThen(testIndexerBackwardsCommand())
+        .andThen(testGamePieceManipulatorForwardsCommand())
+        .andThen(testGamePieceManipulatorBackwardsCommand())
+        .andThen(testClimbCommand())
+        .andThen(testAlgaeRollersForwardsCommand())
+        .andThen(testAlgaeRollersBackwardsCommand())
+        .andThen(testAlgaeIntakeUpCommand())
+        .andThen(testAlgaeIntakeDownCommand())
+        .andThen(testArmElevatorCommand())
+        .andThen(testArmCommand());
+
+  }
+
+  private Command testIndexerForwardsCommand() {
+    return run(() -> indexerSubsystem.runBelt(INDEXER_TESTING_SPEED), indexerSubsystem)
+        .until(indexerSubsystem::isIndexerAtSpeed)
+        .withTimeout(Seconds.of(5))
+        .andThen(this::updateIndexerForwardsTestResult)
+        .finallyDo(indexerSubsystem::stop);
+  }
+
+  private Command testIndexerBackwardsCommand() {
+    return run(() -> indexerSubsystem.runBelt(INDEXER_BACKWARDS_TESTING_SPEED), indexerSubsystem)
+        .until(indexerSubsystem::isIndexerAtSpeed)
+        .withTimeout(Seconds.of(5))
+        .andThen(this::updateIndexerBackwardsTestResult)
+        .finallyDo(indexerSubsystem::stop);
+  }
+
+  private Command testGamePieceManipulatorForwardsCommand() {
+    return run(
+        () -> gamePieceManipulatorSubsystem.runManipulatorWheels(MANIPULATOR_TESTING_SPEED),
+          gamePieceManipulatorSubsystem)
+        .until(gamePieceManipulatorSubsystem::isManipulatorAtSpeed)
+        .withTimeout(Seconds.of(5))
+        .andThen(this::updateManipulatorForwardsTestResult)
+        .finallyDo(gamePieceManipulatorSubsystem::stop);
+  }
+
+  private Command testGamePieceManipulatorBackwardsCommand() {
+    return run(
+        () -> gamePieceManipulatorSubsystem.runManipulatorWheels(MANIPULATOR_BACKWARDS_TESTING_SPEED),
+          gamePieceManipulatorSubsystem)
+        .until(gamePieceManipulatorSubsystem::isManipulatorAtSpeed)
+        .withTimeout(Seconds.of(5))
+        .andThen(this::updateManipulatorBackwardsTestResult)
+        .finallyDo(gamePieceManipulatorSubsystem::stop);
+  }
+
+  private Command testClimbCommand() {
+    return run(() -> testClimbMotors(), climbSubsystem).until(climbSubsystem::isAtClimbVoltage)
```

Comment:
The climb test should probably test if the climb motors are moving (encoder values are increasing/decreasing), maybe do one rotation or something. Testing if the output matches the commanded output doesn't really tell us much.